server/src/main/scala/manifests/actors/RegisteredArrivalsActor.scala

```diff
@@ -0,0 +1,98 @@
+package manifests.actors
+
+import actors.{GetState, PersistentDrtActor, RecoveryActorLike, Sizes}
+import com.trueaccord.scalapb.GeneratedMessage
+import drt.shared.{ArrivalKey, SDateLike}
+import org.slf4j.{Logger, LoggerFactory}
+import server.protobuf.messages.RegisteredArrivalMessage.{RegisteredArrivalMessage, RegisteredArrivalsMessage}
+import services.graphstages.Crunch
+
+
+case class RegisteredArrivals(arrivals: Map[ArrivalKey, Option[Long]])
+
+class RegisteredArrivalsActor(val initialSnapshotBytesThreshold: Int,
+                              now: () => SDateLike,
+                              expireAfterMillis: Long,
+                              val initialMaybeSnapshotInterval: Option[Int]) extends RecoveryActorLike with PersistentDrtActor[RegisteredArrivals] {
+  override def persistenceId: String = "registered-arrivals"
+
+  override def initialState: RegisteredArrivals = RegisteredArrivals(Map())
+
+  override val log: Logger = LoggerFactory.getLogger(getClass)
+
+  override val snapshotBytesThreshold: Int = Sizes.oneMegaByte
+
+  override var state = RegisteredArrivals(Map())
+
+  override def stateToMessage: GeneratedMessage = arrivalsToMessage(state.arrivals)
+
+  private def arrivalsToMessage(arrivalWithLastLookup: Map[ArrivalKey, Option[Long]]): RegisteredArrivalsMessage = {
+    RegisteredArrivalsMessage(
+      arrivalWithLastLookup
+        .map { case (ArrivalKey(o, v, s), l) => RegisteredArrivalMessage(Option(o), Option(v), Option(s), l) }
```

Comment:
do you need to stick something in the `akka.conf` serialisation settings for this?

Reply:
oh shit yes!

Reply:
done. nice catch!